KEYCLOAK-12729 Add password policy not-email

thomasdarimont opened the pull request:
Added test cases and initial translations

<!---
Please read https://github.com/keycloak/keycloak/blob/master/CONTRIBUTING.md and follow these guidelines when contributing to Keycloak
-->

thomasdarimont committed:
KEYCLOAK-12729 Add password policy not-email

Added test cases and initial translations

thomasdarimont commented:
The tests work from me locally - perhaps I'm missing a cleanup procedure after the test run.

mposolda commented on testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/forms/RegisterTest.java:
```diff
@@ -491,6 +491,28 @@ public void registerUserNotUsernamePasswordPolicy() {
+    public void registerUserNotEmailPasswordPolicy() {
+
+        RealmRepresentation notEmailRealm = RealmBuilder.create().passwordPolicy("notEmail").build();
+        notEmailRealm.setRegistrationEmailAsUsername(true);
```
@thomasdarimont As you pointed, you are missing cleanup of those things after the test method.

mposolda commented on testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/account/AccountFormServiceTest.java:
```diff
@@ -412,6 +413,23 @@ public void changePasswordWithNotUsernamePolicy() {
 
+    @Test
+    public void changePasswordWithNotEmailPolicy() {
+        setPasswordPolicy("notEmail(1)");
```
I guess this probably worth cleanup as well.

mposolda reviewed (changes requested):
@thomasdarimont Thanks for the contribution! It looks good to me. Just 2 very minor things:
- Cleanup will be probably needed in some tests as you already pointed in some comment above
- It will be good to send small PR to the documentation as well to make sure this page is updated with the new policy https://www.keycloak.org/docs/latest/server_admin/index.html#_password-policies

mposolda commented:
@thomasdarimont Any update regarding this PR?

mposolda commented:
@thomasdarimont Hi, any update on this PR or are you in the middle of PTO season? :-)

thomasdarimont committed:
KEYCLOAK-12729 Revise password policy not-email tests

- Added missing cleanup to RegisterTest
- Revised test-setup for AccountFormServiceTest

mposolda reviewed (approved):
@thomasdarimont Cool, Thanks! :-)
